### DISCUSSION TITLE
Models not working with RAG UNLESS imported via Openwebui's experimental feature (but broken response)

### DISCUSSION THREAD

**Discussion by antonio-lamberti** [open-webui/open-webui#6299] [General]
### Discussed in https://github.com/open-webui/open-webui/discussions/6297

Imported models, either with the new ollama pull [hf.co/bartowski/Hermes-3-Llama-3.1-8B-GGUF:Q6_K_Lor](http://hf.co/bartowski/Hermes-3-Llama-3.1-8B-GGUF:Q6_K_Lor) via pulling a model on Ollama webside, result in models NOT being able to properly retrieve RAG context in openwebui.
If, instead, I DOWNLOAD locally the gguf file and THEN I import it in openwebui via the experimental feature, it works with RAG, but the response has bad formatting (lots of <|end_of_text|><|begin_of_text|>://->}<|end_of_text|><|begin_of_text|>://->}
<br>  

Why on earth is that?!

![image](https://github.com/user-attachments/assets/01d17b3c-6893-4683-9db4-4c29f5076554)
The 1st one I imported via experimental feature, downloading first the gguf from hf...the 2nd one is downloaded either via pull from Ollama website or hf directly

![image](https://github.com/user-attachments/assets/1aa2ee1c-90b5-4f9d-91b9-47fbf68c88c7)
Model imported via experimental feature has good response, but the formatting is totally bad...

![image](https://github.com/user-attachments/assets/a636b6a4-74fb-4c6f-99b6-5fc786bdbc85)
Models imported via pull (either from ollama or with the new hf method) have no idea of the context, although it's correctly passed to the llm.

Ps. Modelfile content when importing via experimental feature is like this:
TEMPLATE """{{ .System }}
USER: {{ .Prompt }}
ASSISTANT: """
PARAMETER num_ctx 4096
PARAMETER stop "</s>"
PARAMETER stop "USER:"
PARAMETER stop "ASSISTANT:"
</div>

I have then tried to use, in my example, params + template as the modelfile from the website https://ollama.com/taozhiyuai/hermes-2-theta-llama-3:8b-q8_0 but the results is the same as importing directly from HF / Ollama: the llm seems unaware of the context, that is instead being properly retrieved.

**Comment by antonio-lamberti** [comment]
Example, I have tested the model Phi-3.5-mini: importing it from Ollama makes the model UNAWARE of the context.

I then tried to import the model bartowski/Phi-3.5-mini-instruct-GGUF via "ollama pull hf.co/bartowski/Phi-3.5-mini-instruct-GGUF:Q8_0" - this generates this hash file in the blobs ollama folder:

{{ if .System }}<|system|>
{{ .System }}<|end|>
{{ end }}{{ if .Prompt }}<|user|>
{{ .Prompt }}<|end|>
{{ end }}<|assistant|>
{{ .Response }}<|end|>

However, using this model with RAG knowledge in openwebui does not work: the model is not aware of the context.

If I cange it to:

user
{{ if .System }}{{ .System }} {{ end }}{{ .Prompt }}
model
{{ .Response }}

It is RAG context aware, but send endless response. Why is this? How can this be fixed? Is this a bug? It's happening with the vast majority of models.

**Comment by Zayuki** [comment]
I also encounter this error, I pulled Mistral 7b GGUF model from HuggingFace via Ollama. And it does not work well in RAG, seems like not detecting documents at all, whereas other Ollama models are able to read the docs.
